### PR TITLE
Fix `complex` support with CHPL_LOCALE_MODEL=gpu

### PR DESCRIPTION
**Pull request by jabraham17**
Fixes an issue where complex math functions could not be called when using the GPU locale model. This does not add support for complex numbers in GPU kernels.

The root cause of this was due to building the runtime with a C++ compiler, which may not provide the C functions that Chapel uses. This PR builds off of https://github.com/chapel-lang/chapel/pull/25021 to make this work using `__builtin_` functions.

Resolves https://github.com/chapel-lang/chapel/issues/26019
Resolves https://github.com/chapel-lang/chapel/issues/25610

Note: the test added in this PR requires https://github.com/chapel-lang/chapel/pull/26047

Testing
- [x] paratest with/without comm with CPU
- [x] `start_test test/gpu/native` with NVIDIA
- [x] spot check AMD

[Reviewed by @e-kayrakli]

### CODE REVIEW

File: test/gpu/native/complexFunctions.chpl [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// ensure that complex functions and math can be done using the GPU locale model
+// using complex in a kernel is tested by `test/gpu/native/mathOps.chpl`
```
Comment (e-kayrakli) [inline test/gpu/native/complexFunctions.chpl:2]: Shouldn't you have uncommented the pertinent lines in the `mathOps` test?

Comment (jabraham17) [reply test/gpu/native/complexFunctions.chpl:2]: No, because `mathOps` is about `complex` in a kernel. Which still doesn't work even with this PR

Comment (e-kayrakli) [reply test/gpu/native/complexFunctions.chpl:2]: Oh, OK. I remembered you mentioning that maybe GPU support for complex would just fall out from this fix, so seeing this comment made me think that that was the case.

It is OK if it isn't. But could you clarify the comment? If you want, you can just mention that this test is to make sure that the complex functions work on CPU without referring to this test.

It'd be also good to move this test under `test/gpu/nongpu`.

**Review by e-kayrakli — APPROVED** [review]
Thanks for the fix!

**Comment by bradcray** [comment]
Nice!  Thanks Jade!

**Pull request merged**